manager/integration/tests/common.py

```diff
@@ -3337,3 +3339,36 @@ def settings_reset():
 
     client = get_longhorn_api_client()
     reset_settings(client)
+
+
+def wait_for_instance_manager_desire_state(client, core_api, im_name,
+                                           state, desire=True):
+    time.sleep(2 * RETRY_INTERVAL)
+    for i in range(RETRY_COUNTS):
+        im = client.by_id_instance_manager(im_name)
+        try:
+            pod = core_api.read_namespaced_pod(name=im_name,
+                                               namespace=LONGHORN_NAMESPACE)
+        except Exception as e:
+            # Continue with pod restarted case
+            if e.reason == "Not Found":
+                time.sleep(RETRY_INTERVAL)
+                continue
+            # Report any other error
+            else:
+                print("Failed to get pod based on name: %s, namespace: %s"
```

Comment:
Do we expect any other exception to be raised? Sounds like we should `assert` then fail the test.

Reply:
No, we don't expect any other exception here. I will add the assert there to fail the test.

Reply:
Don't need this print line. `assert` will pinpoint the line as well as show up the content of `e`.